lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb

```diff
@@ -1006,6 +1006,12 @@ def cmd_reg(*args)
             end
           end
 
+          if type == 'REG_BINARY'
+            # Use the same format accepted by REG ADD:
+            # REG ADD HKLM\Software\MyCo /v Data /t REG_BINARY /d fe340ead
+            data = data.scan(/../).map {|v| v.to_i(16)}.pack('C*')
```

Comment:
Do we need to consider odd-length strings here? 👀 

Happy path:
```
3.0.2 :003 > 'FF'.scan(/../).map {|v| v.to_i(16)}.pack('C*')
 => "\xFF" 
```

Using `1` or `0`:
```
3.0.2 :004 > '0'.scan(/../).map {|v| v.to_i(16)}.pack('C*')
 => "" 
3.0.2 :005 > '1'.scan(/../).map {|v| v.to_i(16)}.pack('C*')
 => "" 
```

Reply:
Regex on this could also be a lot tighter. I was looking up what the max size of a REG_BINARY data entry can be but unfortunately it doesn't look like there is a max size. https://docs.microsoft.com/en-us/windows/win32/sysinfo/registry-storage-space mentions there is no strict limit however memory shouldn't be charged more than 4 MB on modern systems and that anything over 2KB is strongly encouraged to be written to disk and then referred to by the registry key, however there is nothing preventing a developer from not following this advice and doing whatever they like (which basically means someone out there is doing this cause I've seen developers not follow standards all the time 🤣 )

Here is my suggestion for an improved version for the regex portion of this.

`^[a-zA-Z0-9]{1,}$`

This should now work as shown below and should also have the added benefit of ensuring we only match on valid Hex characters.

```
3.0.2 :009 > 'F'.scan(/^[a-zA-Z0-9]{1,}$/).map {|v| v.to_i(16)}.pack("C*")
 => "\x0F" 
3.0.2 :010 > 
```

Reply:
Resolved in https://github.com/rapid7/metasploit-framework/pull/16479/commits/7aa36a65d6735b8a80bc10b84ca07c4fedb8f1d6 with a different regex that now solves this issue properly.